pkg/option/config.go

```diff
@@ -3113,8 +3114,23 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+
+				// validate the config-map
+				var msgError = "Incorrect config-map flag"
+				for key, value := range m {
+					if val := fmt.Sprintf("%v", value); val != "" {
+						flags := cmd.Flags()
+						// check whether the flag exists
+						if flag := flags.Lookup(key); flag != nil {
```

Comment:
> If `flag` is not found, it should be an error as well, right?

No. The validation works for only existing flags. In case of unknown flags Cilium will start with these unknown flags.